src/db/read.rs

```diff
@@ -36,6 +35,28 @@ impl<'a> OwnedRead<'a> {
     }
 }
 
+pub async fn read_commit(
```

Comment:
Feedback here is similar to the feedback about `from_head()`: I wish this function did less, or that there was another function that just returned the hash and commit. Returning the hash makes sense bc the commit might be from a head. But what if i don't want the map because i'm just hunting for a commit with a particular last mutation id?

I think we should prefer small, orthogonal api calls that each do one thing and then let the caller compose. (Or if it is a common case eg here to also want the map with the commit then we can provide a free fn like this.)

Reply:
This function is internal to the module, it's an implementation detail of read.rs and write.rs.

Reply:
ah, i expected a way to read from head or hash on Commit and saw it was pub and assumed this was the thing i should use to traverse the chain. instead i think I should use Read::get_head to get the head hash then Commit::from_hash to read the commit?